biglog/segment.go

```diff
@@ -164,6 +173,9 @@ func loadSegment(indexPath string) (seg *segment, err error) {
 	seg.setNextOffsets()
 	seg.setCreatedTS()
 
+	// TODO: when this is called via createSegment the index is
+	// created by createSegIndex which doesn't write this timestamp?!
```

Comment:
I had the following idea: What do you think of having a dedicated segment header. It could contain metadata information such as the actual creation date. Other useful information I could come up with is a schema number to make it possible to implement backwards incompatible changes at a later point of time. Maybe it would also be useful to store the value of `maxIndexEntries` upon creation to detect cut off/corrupted index files. Other uses might come up in the future.

Its just a quick idea and we would certainly have to discuss this in more detail in a separate issue. 

Another idea I had was to always create an empty entry in the index when it is created. Could we just run `s.updateIndex(1, 0)` at the very end of `createSegIndex` ?